x/feedist/msg.go

```diff
@@ -0,0 +1,65 @@
+package feedist
+
+import (
+	"fmt"
+
+	"github.com/iov-one/weave/errors"
+)
+
+const (
+	pathNewRevenueMsg    = "feedist/newrevenue"
+	pathDistributeMsg    = "feedist/distribute"
+	pathUpdateRevenueMsg = "feedist/updateRevenue"
+)
+
+func (msg *NewRevenueMsg) Validate() error {
+	if err := msg.Admin.Validate(); err != nil {
+		return errors.Wrap(err, "invalid admin address")
+	}
+	if len(msg.Recipients) == 0 {
+		return errors.InvalidMsgErr.New("at least one recipient must be given")
+	}
+	for i, r := range msg.Recipients {
+		if err := r.Address.Validate(); err != nil {
```

Comment:
q: should we prevent duplicate addresses? We could fail or merge the weights

Reply:
Good idea.
I think either solution is fine, but a check would be good. 

We could also always store then sorted by address and not in order if transaction to make it more deterministic

Reply:
Addresses duplication is no longer allowed. Implemented in https://github.com/iov-one/weave/pull/316/commits/7f699ed7f68580c219ff635f4cd6dc15f5c17b86